src/ui/Templates/TemplateCache.ts

```diff
@@ -35,12 +36,28 @@ export class TemplateCache {
     if (publicTemplate && !_.contains(TemplateCache.templateNames, name)) {
       TemplateCache.templateNames.push(name);
     }
+
+    if (pageTemplate && !_.contains(TemplateCache.pageTemplateNames, name)) {
+      TemplateCache.pageTemplateNames.push(name);
+    }
+
     if (defaultTemplate) {
       TemplateCache.defaultTemplates[name] = template;
     }
   }
 
   /**
+   * Register a new page template in the framework. Equivalent to put script tags in the DOM.
+   * @param name
+   * @param template
+   */
+  public static registerPageTemplate(name: string, template: Template);
```

Comment:
I see... 

Can't we instead refactor the registerTemplate method to accept a configuration object ? Instead of a long list of parameters ?

In any case, I would not document this publicly. 

I don't see why someone doing an integration would need to see this in the reference documentation in order to use our stuff.

Nor would I think they would actually understand **what** it's used for, and why.